x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -432,7 +439,22 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                 // Endpoint metrics. Kibana requires read access to send telemetry
                 RoleDescriptor.IndicesPrivileges.builder()
                     .indices("metrics-endpoint.metrics-*")
-                    .privileges("read").build()
+                    .privileges("read").build(),
+                // Fleet package upgrade
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("logs-*", "synthetics-*", "traces-*",
+                        "/metrics-.*&~(metrics-endpoint\\..*)/")
```

Comment:
@joshdover The index pattern `metrics-*` overlap with a few index patterns beginning with `metrics-endpoint.`: 
* `metrics-endpoint.policy-*`
* `metrics-endpoint.metrics-*`
* `metrics-endpoint.metadata*`
* `metrics-endpoint.metadata_current_default`
* `metrics-endpoint.metadata_united_default`

But privileges required for the above are different from those required for `metrics-*`. For example, `create_index` is required for `metrics-*` but not the others. On the other hand, `read` is required for `metrics-endpoint.metadata_current_default` but not the more general `metrics-*`. 

I suspect that we do not want to merge all these privileges under the more general `metrics-*` pattern. Hence this line uses Lucene regex to grant privileges to `metrics-*` **except** `metric-endpoint.*`. And the individual `metric-endpoint.*` index patterns are handled separately.

Please let me know whether it would work for the current design.

Reply:
Agreed we can limit this to avoid giving all these privileges to the transform destination indices. We will still need `create_index` for these though as noted below.

However, I think we do still need these privileges for indices like `metrics-endpoint.policy-*` and `metrics-endpoint.metrics-*`? For reference, the data streams that Kibana will be creating for endpoint is specified in this package spec here: https://github.com/elastic/package-storage/tree/production/packages/endpoint/1.1.1/data_stream